Implement Uploader

raddmag opened the pull request:
Did not use apache module.
Although it takes in a LocationJSON, it’s basically an abstract
uploader for JSON’s.
Might need to fix LocationJSON so that it stores them in a map.
Did not check for getActiveNetworkInfo() or isConnected(), might be
better to check before call, if not easy fix

raddmag committed:
Implement Uploader

Did not use apache module.
Although it takes in a LocationJSON, it’s basically an abstract
uploader for JSON’s.
Might need to fix LocationJSON so that it stores them in a map.
Did not check for getActiveNetworkInfo() or isConnected(), might be
better to check before call, if not easy fix

kmh287 commented on app/src/main/java/cs5150athletetracking/com/AsyncUploader/AsyncUploader.java:
```diff
@@ -0,0 +1,89 @@
+        /* Sent LocationJson. Check for received confirmation */
+
+        if(result == 1){
+            //ResultCallable.success();
```
How does the callback get passed into this object?

kmh287 commented on app/src/main/java/cs5150athletetracking/com/AsyncUploader/AsyncUploader.java:
```diff
@@ -0,0 +1,89 @@
+    private static final int HTTP_OK = 200;
+
+    @Override
+    protected Integer doInBackground(LocationJSON... params) {
```
abstract to JSONObject

kmh287 commented:
Looks good, just a few issues to resolve. Please see comments.

raddmag committed:
Change params to JSONObject and implement ResultCallable

Changed AsyncUploader to send up all params.
Changed AsyncUploader to implement ResultCallable, not sure the
functions should do.
Alternatively, could make ResultCallable a class rather than an
interface as it was initially designed.

kmh287 commented on app/src/main/java/cs5150athletetracking/com/athletetracking/AsyncUploader.java:
```diff
@@ -0,0 +1,102 @@
+public class AsyncUploader extends AsyncTask<JSONObject, String, Integer> implements ResultCallable {
+
+    private static final String TAG = "Http Connection";
+    private static final String urlStr = "http://ec2-54-165-208-160.compute-1.amazonaws.com/php/test_server.php";
```
static variables should be in SCREAMING_SNAKE_CASE

kmh287 commented on app/src/main/java/cs5150athletetracking/com/athletetracking/AsyncUploader.java:
```diff
@@ -0,0 +1,102 @@
+ * Created by mag94 on 10/19/15.
+ */
+
+public class AsyncUploader extends AsyncTask<JSONObject, String, Integer> implements ResultCallable {
```
I suppose this could work.

So when I create this object, I'll need to override these methods, right?

raddmag replied:
Yea I guess we could do that. Unless making callable an object that can get passed in as a parameter, in that case I would have to add a constructor. This might be better, unless you want to stick with interace

kmh287 replied:
I think it might be best to just have a setter for a callback and then later on call it (if the callback isn't null)

raddmag replied:
Alright, I'll make the chsnges.

kmh287 commented on app/src/main/java/cs5150athletetracking/com/athletetracking/AsyncUploader.java:
```diff
@@ -0,0 +1,102 @@
+
+        try {
+            int count = params.length;
+            for (int i = 0; i < count; i++) {
```
Don't think we'll ever pass in more than one param, but this is good thinking just in case.

kmh287 commented on app/src/main/java/cs5150athletetracking/com/athletetracking/AsyncUploader.java:
```diff
@@ -0,0 +1,102 @@
+                        message += (char)c;
+
+                    Log.d(TAG, message);
+                    result = 1;
```
Create constants for these

kmh287 commented on app/src/main/java/cs5150athletetracking/com/athletetracking/AsyncUploader.java:
```diff
@@ -0,0 +1,102 @@
+                    Log.d(TAG, message);
+                    result = 1;
+                } else {
+                    result = 0;
```
constant

kmh287 commented on app/src/main/java/cs5150athletetracking/com/athletetracking/AsyncUploader.java:
```diff
@@ -0,0 +1,102 @@
+    protected void onPostExecute(Integer result) {
+        /* Sent LocationJson. Check for received confirmation */
+
+        if(result == 1){
```
constant

kmh287 commented on app/src/main/java/cs5150athletetracking/com/athletetracking/ResultCallable.java:
```diff
@@ -0,0 +1,10 @@
+package cs5150athletetracking.com.athletetracking;
+
+/**
+ * Created by mag94 on 10/20/15.
```
I merged my result callable into staging. If it isn't showing up for you on your branch, pull in from staging.

kmh287 commented on app/src/main/java/cs5150athletetracking/com/athletetracking/AsyncUploader.java:
```diff
@@ -0,0 +1,102 @@
+
+                // Format LocationJSON argument for the POST method
+                // The format is: "key1=value1&key2=value2"
+                while (keys.hasNext()) {
```
Is this going to be able to handle JSONObjects with arbitrary structure? Our location data is probably going to be a JSONObject that contains a JSONArray of JSONObjects

raddmag replied:
I thought that it was just going to be JSON object, this allows you to store different types that aren't necessarily string types. What is the benefit of the other way

kmh287 replied:
JSONs ARE strings.
